File: app/src/main/java/com/xoleric/music/core/model/Album.kt
package com.xoleric.music.core.model

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "albums",
    indices = [Index(value = ["mediaStoreId"], unique = true)]
)
data class Album(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val mediaStoreId: Long,
    val title: String,
    val artist: String,
    val artistId: Long = 0,
    val albumArtUri: String? = null,
    val songCount: Int = 0,
    val year: Int = 0,
    val dateAdded: Long = 0
)

File: lib/models/album.dart
class AlbumModel {
  final int id;
  final String title;
  final String artist;
  final int? artworkId;
  final int songCount;

  const AlbumModel({
    required this.id,
    required this.title,
    required this.artist,
    this.artworkId,
    required this.songCount,
  });
}
